### PR TITLE
refactor(ui/config): Modify the configuration file read policy

### PR DESCRIPTION
**Pull request by miRemid**
## Proposed Changes

- Modify the configuratiuon file read policy

Now, KubeSpider will check whether the configuration file exists. If it does not exist, it will automatically create a default configuration file in the specified directory and run it. The specified directory can be specified through the environment variable `KUBESPIDER_HOME`. The default path is `/data/kubespider`

For local developer, there is no need to conflict with the original system `HOME` env var.
```
KUBESPIDER_HOME=$HOME/.config/kubespider python kubespider/app.py
```

Subsequent sqlite files and source_provider bins will be uniformly saved in the KUBESPIDER_HOME directory.
```
❯ tree $KUBESPIDER_HOME
/root/.config/kubespider
├── kubespider.db
├── kubespider.yaml
└── providers
    └── source_bin
        └── mikanani_provider

3 directories, 3 file
```

- Support for env vars override

Now, KubeSpider supports reading environment variables to overwrite configurations read from `$KUBESPIDER_HOME/kubespider.yaml`, which facilitates container startup and local debugging.

- Add logging level config

Now, KubeSpider supports setting the logging level from the `$KUBESPIDER_HOME/kubespider.yaml` or env var `KUBESPIDER_LOG_LEVEL` to change the logging mode, only support for "DEBUG" and "INFO" 

```
# $HOME/.config/kubespider/kubespider.yaml
...
log:
    level: INFO
webapi:
    server_port: 3080
...

> KUBESPIDER_HOME=$HOME/.config/kubespider \
  KUBESPIDER_WEBAPI_SERVER_PORT=4080 \
  KUBESPIDER_LOG_LEVEL=DEBUG \
  python kubespider/app.py

2023-12-25 19:19:50,705-INFO runner.py[line11]: Webhook Server start running...
2023-12-25 19:19:50,724-INFO manager.py[line38]: [DownloadManager] instance reload success ...
2023-12-25 19:19:50,726-INFO manager.py[line28]: [SourceManager] instance reload success ...
2023-12-25 19:19:50,727-INFO manager.py[line47]: [NotificationManager] instance reload success ...
2023-12-25 19:19:50,738-DEBUG selector_events.py[line54]: Using selector: EpollSelector
2023-12-25 19:19:50,740-INFO wasyncore.py[line486]: Serving on http://0.0.0.0:4080
```

### CODE REVIEW

File: kubespider/utils/global_config.py [hunk 1]
```diff
@@ -1,26 +1,74 @@
...
+import os
+import os.path as osp
+from enum import Enum
+
+import yaml
+
 from utils.config_reader import YamlFileConfigReader
-from utils.values import CFG_BASE_PATH
 
+CFG_BASE_PATH = os.getenv('KUBESPIDER_HOME', '/data/kubespider')
```
Comment (evell1992) [inline kubespider/utils/global_config.py:10]: /data/kubespider 这个路径有没有其他地方用到,这个可能要和@cesign大佬确认下,我觉得还是不要改的好.之前应该是个.config文件夹

File: kubespider/utils/global_config.py [hunk 2]
```diff
@@ -1,26 +1,74 @@
...
+  base_path: /downloads
+"""
 
+class PathConfig(str, Enum):
+    KUBESPIDER_CONFIG = 'kubespider.yaml'
+    DEPENDENCIES_CONFIG = 'dependencies/'
+    SOURCE_PROVIDERS_BIN = 'providers/source_bin'
+    SOURCE_PROVIDERS_CONF = 'providers/source'
+    DOWNLOAD_PROVIDERS_CONF = 'providers/download'
+    NOTIFICATION_PROVIDERS_CONF = 'providers/notification'
```
Comment (evell1992) [inline kubespider/utils/global_config.py:34]: SOURCE_PROVIDERS_CONF DOWNLOAD_PROVIDERS_CONF  NOTIFICATION_PROVIDERS_CONF 这三个我晚点改完配置存储再删

File: kubespider/utils/global_config.py [hunk 3]
```diff
@@ -1,26 +1,74 @@
...
-cfg = get_global_config().read()
+    def __str__(self) -> str:
+        return str(self.value)
+
+    def config_path(self) -> str:
+        return os.path.join(CFG_BASE_PATH, self)
 
 
+def get_global_config() -> YamlFileConfigReader:
+    if not osp.exists(PathConfig.KUBESPIDER_CONFIG.config_path()):
```
Comment (evell1992) [inline kubespider/utils/global_config.py:44]: initdirs调用的位置是在这个方法调用之后,所以可能会存在文件夹不存在的情况

File: kubespider/core/runner.py [hunk 4]
```diff
@@ -4,16 +4,10 @@
...
 
 from core.api import create_app
-from core import config_handler
-from utils.global_config import Config
+from utils.global_config import Config, initdirs
 
 
 def run() -> None:
     logging.info('Webhook Server start running...')
+    initdirs()
```
Comment (evell1992) [inline kubespider/core/runner.py:12]: 在调用这个方法之前, get_global_config 已经被调用过了,这个方法可能需要前置些.

File: kubespider/utils/config_reader.py [hunk 5]
```diff
@@ -1,9 +1,117 @@
+import logging
 import threading
 import os
 from abc import ABC, abstractmethod
 
 import yaml
 
+class StructConfig:
```
Comment (evell1992) [inline kubespider/utils/config_reader.py:8]: 这个类是一个新的数据结构用来代替dict的吗

Comment (miRemid) [reply kubespider/utils/config_reader.py:8]: 是的，多了注释里面特性

File: kubespider/utils/global_config.py (hunk 3 above)
Comment (miRemid) [reply kubespider/utils/global_config.py:43]: Done，现在把初始化放在这里了，可以保证文件夹一定存在

File: kubespider/core/runner.py (hunk 4 above)
Comment (miRemid) [reply kubespider/core/runner.py]: Done

**Pull request merged**